htdocs/core/class/html.form.class.php

```diff
@@ -3387,6 +3387,14 @@ public function select_produits_fournisseurs_list($socid, $selected = '', $htmln
 				$opt .= $optlabel;
 				$outval .= $outvallabel;
 
+				$hookmanager->initHooks(array('ordersuppliercard'));
```

Comment:
The initHooks must be done by the main page. 
So when select_produits_fournisseurs_list is called by page a, context is a, when called by page b, context os b.